### ISSUE TITLE
An issue in test_production.py: sess.run([pred_boxes, pred_confidences], feed_dict=feed) hangs and no response.

### ISSUE THREAD

**Issue by Chenyangzh** [Russell91/TensorBox#305]
Hello, I am new and  trying to run 'test_production.py' to test other images which are not belongs to the brainwash-dataset. I trained a model using 'overfeat_rezoom.json' scrips and get a result of .ckpt files. When I tried to run test file an error came out "can't find placeholders" in the line of 

`x_in = tf.get_collection('placeholders')[0]`

I found some tutorials on internet, know that I need to 'add_to_collection' to collect tensors as a list. But I'm not sure where I need to add. Not only the input node 'x_in', but also the output nodes 'pred_boxes, pred_confidences'. But I tried many times, by adding the arguments in model.py as follow:

After the placeholder maker in in train()
`x_in = tf.placeholder(tf.float32)` 
or like this in build()
`x, confidences, boxes = q[phase].dequeue_many(arch['batch_size'])
            tf.add_to_collection('placeholders', x[0])`
and put output-collections like this in build():
`(pred_boxes, pred_confidences,
             loss[phase], confidences_loss[phase],
             boxes_loss[phase]) = self.build_forward_backward(x, phase, boxes, flags)
            tf.add_to_collection('vars', pred_boxes)
            tf.add_to_collection('vars', pred_confidences)`

I am not sure if it is right. And finally, I ran the test_production.py with a crazy error. There is no warnings but can not go over the sess.run() line:
`(np_pred_boxes, np_pred_confidences) = sess.run([pred_boxes, pred_confidences], feed_dict=feed)`

Anyone can give some advices to me? Thanks a lot!
 Follows are some information after I run the test_production.py code. ( I also get same things when running train.py before it enter into normal training iterations.)

`2019-03-14 19:26:00.910245: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-03-14 19:26:01.013789: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:964] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-03-14 19:26:01.014330: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1432] Found device 0 with properties: 
name: GeForce RTX 2080 major: 7 minor: 5 memoryClockRate(GHz): 1.71
pciBusID: 0000:01:00.0
totalMemory: 7.76GiB freeMemory: 7.26GiB
2019-03-14 19:26:01.014340: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1511] Adding visible gpu devices: 0
2019-03-14 19:26:01.227351: I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-03-14 19:26:01.227373: I tensorflow/core/common_runtime/gpu/gpu_device.cc:988]      0 
2019-03-14 19:26:01.227377: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1001] 0:   N 
2019-03-14 19:26:01.227586: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 6986 MB memory) -> physical GPU (device: 0, name: GeForce RTX 2080, pci bus id: 0000:01:00.0, compute capability: 7.5)
WARNING:tensorflow:From /home/ai_001/TensorflowPJ/TensorBox-master/test_production.py:55: start_queue_runners (from tensorflow.python.training.queue_runner_impl) is deprecated and will be removed in a future version.
Instructions for updating:
To construct input pipelines, use the `tf.data` module.
WARNING:tensorflow:`tf.train.start_queue_runners()` was called when no queue runners were defined. You can safely remove the call to this deprecated function.`

**Comment by Chenyangzh** [comment]
Hi, I found the problem here. The reason of hanging is when I restore the ".meta" graph, the queue list cannot create input pipline well for some reason in the sub-thread. Then, the main thread will wait the input image forever... My solution is putting the input `tf.add_to_collection('placeholders')` after the dequeue argument :`x, confidences, boxes = q[phase].dequeue_many(arch['batch_size'])` and the output `tf.add_to_collection('vars')` after the build_forward_backward function `x, confidences, boxes = q[phase].dequeue_many(arch['batch_size'])`. Right now, it can produce images in one thread. But I still don't know how to fix the muilt-thread problem. Anyone know that? Pls let me know.

**Comment by RomanSteinberg** [comment]
@Chenyangzh what version of Tensorflow do you use? Do you use GPU or CPU for inference?

**Comment by lonelylingoes** [comment]
I have the same problem with you. But your solution seams not work with my problem.
I use Tensorflow-gpu==1.9.0 and use gpu for inference.

**Comment by Chenyangzh** [comment]
> @Chenyangzh what version of Tensorflow do you use? Do you use GPU or CPU for inference?

I used tensorflow 1.12.0-gpu
